impl/src/main/java/org/jboss/weld/bean/proxy/ClientProxyProvider.java

```diff
@@ -258,4 +301,49 @@ public void clear() {
         this.requestedTypeClosureProxyPool.clear();
     }
 
+    /**
+     * Creates and returns a new {@link ClientProxyProvider} instance.
+     *
+     * <p>This method never returns {@code null}.</p>
+     *
+     * <p>This method uses the {@link ServiceLoader#load(Class,
+     * ResourceLoader)} method internally to find the appropriate
+     * subclass to instantiate.  If there is a problem instantiating
+     * such a subclass, or if no service provider has been designated
+     * at all, a {@linkplain #ClientProxyProvider(String) new
+     * <code>ClientProxyProvider</code>} is returned instead (the most
+     * common case).</p>
+     *
+     * <p>It is guaranteed that an invocation of the {@link
+     * #getContainerId()} method on the returned {@link
+     * ClientProxyProvider} will return a {@link String} {@linkplain
+     * String#equals(Object) equal to} the supplied {@code
+     * containerId}.</p>
+     *
+     * @param resourceLoader a {@link ResourceLoader} that will be
+     * used by a {@link ServiceLoader}; may be {@code null}
+     *
+     * @param containerId the identifier of the {@link Container} on
+     * behalf of which the new {@link ClientProxyProvider} is being
+     * built; may be {@code null} but behavior of the returned {@link
```

Comment:
I don't think we should allow `null` here.

Reply:
OK; I can add a `null` check. (I was just documenting usage that was already there.)